account_invoice_customer_ref_unique/models/account_invoice.py

```diff
@@ -19,47 +19,34 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ###############################################################################
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
 
-from openerp.osv import orm
-from openerp.tools.translate import _
+import logging
+_logger = logging.getLogger(__name__)
 
 
-class AccountInvoice(orm.Model):
+class AccountInvoice(models.Model):
     _inherit = "account.invoice"
 
-    def copy(self, cr, uid, ids, default=None, context=None):
-        default = default or {}
-        default.update({
-            'name': '',
-        })
-        return super(AccountInvoice, self).copy(cr, uid, ids, default, context)
-
-    def _check_unique_name_insensitive(self, cr, uid, ids, context=None):
-        for i_id in ids:
-            invoice = self.browse(cr, uid, i_id, context=context)
+    @api.multi
+    @api.constrains('name')
+    def _check_unique_name_insensitive(self):
```

Comment:
In order to increase the performance is faster use a sql constraint based on computed fields with store=True

Reply:
check the following example where is 4x faster than a python constraint using too many invoices https://github.com/Vauxoo/addons-vauxoo/blob/8.0/supplier_invoice_number_unique/model/account_invoice.py